crates/task-impls/src/consensus/proposal_helpers.rs

```diff
@@ -1048,11 +1022,36 @@ pub async fn handle_quorum_proposal_validated<TYPES: NodeType, I: NodeImplementa
             };
         }
 
-        ensure!(
-            task_state.vote_if_able(&event_stream).await,
-            "Failed to vote"
-        );
-        task_state.current_proposal = None;
```

Comment:
After reviewing changes in `consensus/mod.rs`, I think it makes sense to remove this line, since we'll do this when handling the `QuorumVoteSend` event. But I also think we should do the same for the dependency tasks.